### PR TITLE
Add TuMangas.net

### LINKED ISSUE #3400: TuMangas.net
### Source name

TuMangas.net

### Source link

https://tumangas.net/

### Source language

Spanish 

### Other details

18+/NSFW = yes

### Acknowledgements

- [X] I have checked that the extension does not already exist by searching the [extension listing](https://keiyoushi.github.io/extensions/) and verified it does not appear there.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a meaningful title with the source name.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by bapeey**
Closes #3400

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

### CODE REVIEW

File: src/es/tumangasnet/src/eu/kanade/tachiyomi/extension/es/tumangasnet/TuMangasNet.kt [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        title = element.selectFirst(".title")!!.text().substringBeforeLast("Ep.").trim()
+        thumbnail_url = element.selectFirst("figure > img")!!.attr("src")
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotBlank()) {
+            val url = "$baseUrl/biblioteca-manga".toHttpUrl().newBuilder()
+                .addQueryParameter("buscar", query)
+                .addQueryParameter("page", page.toString())
+                .toString()
```
Comment (AwkwardPeak7) [inline src/es/tumangasnet/src/eu/kanade/tachiyomi/extension/es/tumangasnet/TuMangasNet.kt]:
```suggestion
                .build()
```

File: src/es/tumangasnet/src/eu/kanade/tachiyomi/extension/es/tumangasnet/TuMangasNet.kt [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    override fun getFilterList() = FilterList(
+        Filter.Header("NOTA: Los filtros no funcionan en la búsqueda por texto."),
+        Filter.Separator(),
+        GenreFilter(),
+    )
+
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        document.selectFirst("article.anime-single")?.let { element ->
```
Comment (AwkwardPeak7) [inline src/es/tumangasnet/src/eu/kanade/tachiyomi/extension/es/tumangasnet/TuMangasNet.kt]:
```suggestion
        document.selectFirst("article.anime-single")!!.let { element ->
```

File: src/es/tumangasnet/src/eu/kanade/tachiyomi/extension/es/tumangasnet/TuMangasNet.kt [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            title = element.selectFirst(".title")!!.text()
+            genre = element.select("p.genres > span").joinToString { it.text() }
+            description = element.selectFirst(".sinopsis")!!.text()
+            thumbnail_url = element.selectFirst("div.thumb figure > img")!!.attr("abs:src")
```
Comment (AwkwardPeak7) [inline src/es/tumangasnet/src/eu/kanade/tachiyomi/extension/es/tumangasnet/TuMangasNet.kt]:
```suggestion
            description = element.selectFirst(".sinopsis")?.text()
            thumbnail_url = element.selectFirst("div.thumb figure > img")?.attr("abs:src")
```

File: src/es/tumangasnet/src/eu/kanade/tachiyomi/extension/es/tumangasnet/TuMangasNet.kt [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    override fun latestUpdatesSelector() = "ul.episodes article.episode"
+
+    override fun latestUpdatesFromElement(element: Element) = SManga.create().apply {
+        setUrlWithoutDomain(
+            element.selectFirst("a")!!.attr("href")
+                .substringBeforeLast("-")
+                .replace("/leer-manga/", "/manga/"),
+        )
+        title = element.selectFirst(".title")!!.text().substringBeforeLast("Ep.").trim()
+        thumbnail_url = element.selectFirst("figure > img")!!.attr("src")
```
Comment (AwkwardPeak7) [inline src/es/tumangasnet/src/eu/kanade/tachiyomi/extension/es/tumangasnet/TuMangasNet.kt]:
```suggestion
        thumbnail_url = element.selectFirst("figure > img")?.attr("src")
```

File: src/es/tumangasnet/src/eu/kanade/tachiyomi/extension/es/tumangasnet/TuMangasNet.kt [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    override fun searchMangaNextPageSelector() = "ul.pagination li.page-item a[rel=next]"
+
+    override fun searchMangaSelector() = "ul.animes article.anime"
+
+    override fun searchMangaFromElement(element: Element) = SManga.create().apply {
+        setUrlWithoutDomain(element.selectFirst("a")!!.attr("href"))
+        title = element.selectFirst(".title")!!.text()
+        thumbnail_url = element.selectFirst("figure > img")!!.attr("src")
```
Comment (AwkwardPeak7) [inline src/es/tumangasnet/src/eu/kanade/tachiyomi/extension/es/tumangasnet/TuMangasNet.kt]:
```suggestion
        thumbnail_url = element.selectFirst("figure > img")?.attr("src")
```

File: src/es/tumangasnet/src/eu/kanade/tachiyomi/extension/es/tumangasnet/TuMangasNet.kt [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    override fun getFilterList() = FilterList(
+        Filter.Header("NOTA: Los filtros no funcionan en la búsqueda por texto."),
+        Filter.Separator(),
```
Comment (AwkwardPeak7) [inline src/es/tumangasnet/src/eu/kanade/tachiyomi/extension/es/tumangasnet/TuMangasNet.kt]: no need for a seperator imo

File: src/es/tumangasnet/src/eu/kanade/tachiyomi/extension/es/tumangasnet/TuMangasNet.kt [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+    override fun searchMangaFromElement(element: Element) = SManga.create().apply {
+        setUrlWithoutDomain(element.selectFirst("a")!!.attr("href"))
+        title = element.selectFirst(".title")!!.text()
+        thumbnail_url = element.selectFirst("figure > img")?.attr("src")
+    }
+
+    override fun getFilterList() = FilterList(
+        Filter.Header("NOTA: Los filtros no funcionan en la búsqueda por texto."),
+        Filter.Separator(),
```
Comment (bapeey) [inline src/es/tumangasnet/src/eu/kanade/tachiyomi/extension/es/tumangasnet/TuMangasNet.kt]:
```suggestion
```

**Review by AwkwardPeak7 — APPROVED** [review]

**Pull request merged**